pkg/kn/commands/service/update.go

```diff
@@ -135,7 +136,11 @@ func NewServiceUpdateCommand(p *commands.KnParams) *cobra.Command {
 			if waitFlags.Wait && targetFlag == "" {
 				fmt.Fprintf(out, "Updating Service '%s' in namespace '%s':\n", args[0], namespace)
 				fmt.Fprintln(out, "")
-				err := waitForService(cmd.Context(), client, name, out, waitFlags.TimeoutInSeconds)
+				wconfig := clientservingv1.WaitConfig{
+					Timeout:     time.Duration(waitFlags.TimeoutInSeconds) * time.Second,
+					ErrorWindow: time.Duration(waitFlags.ErrorWindowInSeconds) * time.Second,
+				}
+				err := waitForService(cmd.Context(), client, name, out, wconfig)
```

Comment:
`service update` cmd is missing the new flag?

``` 
waitFlags.AddErrorWindow(serviceCreateCommand, "service", "ready")
```

Reply:
It would be resolved by my latter comment, once `--timeout` is part of `AddConditionWaitFlags`.